operator/bucketcontroller/create.go

```diff
@@ -54,6 +57,27 @@ func (p *ProvisioningPipeline) createS3Bucket(ctx *pipelineContext) error {
 	return nil
 }
 
+// enrichConnectionSecret adds more data to the already existing secret such as EndpointName, RegionName and BucketName
+func (p *ProvisioningPipeline) enrichConnectionSecret(ctx *pipelineContext) error {
+	bucket := ctx.bucket
+	log := controllerruntime.LoggerFrom(ctx)
+
+	secretRef := bucket.Spec.ForProvider.CredentialsSecretRef
+
+	credentialSecret := &corev1.Secret{ObjectMeta: metav1.ObjectMeta{Name: secretRef.Name, Namespace: secretRef.Namespace}}
+	_, err := controllerruntime.CreateOrUpdate(ctx, p.kube, credentialSecret, func() error {
+		credentialSecret.Data[EndpointName] = []byte(bucket.Spec.ForProvider.EndpointURL)
+		credentialSecret.Data[RegionName] = []byte(bucket.Spec.ForProvider.Region)
+		credentialSecret.Data[BucketName] = []byte(bucket.GetBucketName())
+		return nil
```

Comment:
Please add an owner reference.
```suggestion
		return controllerruntime.AddOwnerReference(bucket, secret, scheme)
```

Reply:
yep sure

Reply:
(Note that I didn't provide a fully compilable code suggestion, please fix it accordingly)

Reply:
I commit only when I see that it helps me to correct faster. In any case the code is checked and even to commit history is overwritten.